Geometry/HcalAlgo/plugins/DDHCalAngular.cc

```diff
@@ -32,37 +38,44 @@ void DDHCalAngular::initialize(const DDNumericArguments & nArgs,
   n           = int (nArgs["n"]);
   startCopyNo = int (nArgs["startCopyNo"]);
   incrCopyNo  = int (nArgs["incrCopyNo"]);
-  LogDebug("HCalGeom") << "DDHCalAngular debug: Parameters for positioning-- "
-		       << n << " copies in " << rangeAngle/CLHEP::deg 
-		       << " from " << startAngle/CLHEP::deg << "\tShifts " 
-		       << shiftX << ", " << shiftY 
-		       << " along x, y axes; \tZoffest " << zoffset
-		       << "\tStart and inremental copy nos " << startCopyNo 
-		       << ", " << incrCopyNo;
-
+#ifdef EDM_ML_DEBUG
+  edm::LogVerbatim("HCalGeom") << "DDHCalAngular: Parameters for positioning "
+			       << "-- " << n << " copies in " 
+			       << convertRadToDeg(rangeAngle) << " from " 
+			       << convertRadToDeg(startAngle) << "\tShifts " 
+			       << shiftX << ", " << shiftY 
+			       << " along x, y axes; \tZoffest " << zoffset
+			       << "\tStart and inremental copy nos " 
+			       << startCopyNo << ", " << incrCopyNo;
+#endif
   rotns       = sArgs["RotNameSpace"];
   idNameSpace = DDCurrentNamespace::ns();
   childName   = sArgs["ChildName"]; 
+#ifdef EDM_ML_DEBUG
   DDName parentName = parent().name(); 
-  LogDebug("HCalGeom") << "DDHCalAngular debug: Parent " << parentName 
-		       << "\tChild " << childName << "\tNameSpace "
-		       << idNameSpace << "\tRotation Namespace " << rotns;
+  edm::LogVerbatim("HCalGeom") << "DDHCalAngular: Parent " << parentName 
+			       << "\tChild " << childName << "\tNameSpace "
+			       << idNameSpace << "\tRotation Namespace "
+			       << rotns;
+#endif
 }
 
 void DDHCalAngular::execute(DDCompactView& cpv) {
 
   double dphi   = rangeAngle/n;
-  double phi    = startAngle;
+  double phix   = startAngle;
   int    copyNo = startCopyNo;
+  double theta  = 90._deg;
 
   for (int ii=0; ii<n; ii++) {
-
-    double phideg = phi/CLHEP::deg;
+    if      (phix >= 2._pi) phix -= 2._pi;
+    else if (phix <  0)     phix += 2._pi;
+    double phiy   = phix + 90._deg;
+    double phideg = convertRadToDeg(phix);
     int    iphi;
-    if (phideg > 0) iphi = int(phideg+0.1);
-    else            iphi = int(phideg-0.1);
-    if (iphi >= 360) iphi   -= 360;
-    phideg = iphi;
+    if (phideg > 0)  iphi  = int(phideg+0.1);
+    else             iphi  = int(phideg-0.1);
+    if (iphi >= 360) iphi -= 360;
```

Comment:
This line is redundant with line 71 since `iphi` comes from `phideg` which comes from `phix`. This line should be deleted. If you are worried about `phix` originally being greater than 4pi, then lines 71-72 should be fixed to cover that case.

Reply:
I'm working on an angle utility class to handle the operations in lines 71-77. If you can wait on this PR, it could be a test of the new utility, but if you want the PR merged quickly, then don't wait for me.

Reply:
I shall wait for that class for the other algorithms which I need to take care. I am using the rounding utility as suggested by Yana for these two classes